### PR TITLE
STR - 87 -- FE: Display "good" and "bad" research json and ask admin user whether to submit or not

### PR DESCRIPTION
**Pull request by ekno27**
# Description
Part 3 of the 4-part admin series, 
allows the admin to add the 'good' students onto the database by submitting a file and then then confirming 
  
# Testing
1. login as steve
2. use the test_file.xlsx provided in the slack channel
3. the page will show you 'good' and 'bad student data
4. press yes to submit 
## NOTE: the route will throw a 404 similar the the part I of the series as well as a message saying the data was not submitted, this is because the route does not exist in the backend and will be added on str-88.
5. submit the file one more time but this time say no, this will clear the page and display the message
 ## NOTE: The Sequel: styling is not a priority, there is a task attached to that.

### CODE REVIEW

**Review by BrianLinggadjaja — DISMISSED** [review]
Everything works, just remove the semi-colon's 👍

File: resources/js/components/admin/Participants.vue [hunk 1]
```diff
@@ -0,0 +1,61 @@
+<template>
+  <div>
+    <div class="badParticipants" v-if="categorizedParticipants.bad !== undefined">
```
Comment (frlzjosh) [inline resources/js/components/admin/Participants.vue]: how about a class name of `participants__bad` and vice versa for good

File: resources/js/components/admin/Participants.vue [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+export default {
+  name: 'Participants',
+  computed: {
+    ...mapGetters([
+      'categorizedParticipants'
+    ])
+  },
+  methods: {
+    submitGoodParticipants() {
+      console.log(this.categorizedParticipants.good)
```
Comment (frlzjosh) [inline resources/js/components/admin/Participants.vue:53]: left in a console.log(). Take out unless you're saving it for later

File: resources/js/views/admin.vue [hunk 3]
```diff
@@ -8,13 +8,21 @@
     </label>
     <button v-if="!incorrectFileType" @click.prevent="submitFile">Submit</button>
   </div>
+  <div>
+    <Participants v-if="participantsWereSubmitted===null"/>
+    <h2 v-if="participantsWereSubmitted===true">Participants were submitted!</h2>
```
Comment (frlzjosh) [inline resources/js/views/admin.vue]: let's just do a v-else here

File: resources/js/views/admin.vue [hunk 4]
```diff
@@ -25,60 +33,71 @@ export default {
...
+      this.$store.commit('PARTICIPANTS_WERE_SUBMITTED', null);
+      this.$store.dispatch('verifyExcelSheet', this.participants);
     },
     readFile (file) {
       /* Boilerplate to set up FileReader */
-      var reader = new FileReader()
+      var reader = new FileReader();
       reader.onload = (e) => {
-        var bstr = e.target.result
+        var bstr = e.target.result;
```
Comment (frlzjosh) [inline resources/js/views/admin.vue]: `bstr`, `wb`, `wsname`, are unreadable var names

**Review by frlzjosh — CHANGES_REQUESTED** [review]
Issues with some var names, and i gave you advice on things you can do to improve your code.

File: resources/js/components/admin/Participants.vue (hunk 2 above)
Comment (ekno27) [reply resources/js/components/admin/Participants.vue:53]: saving for later

**Review by frlzjosh — APPROVED** [review]
aye nice man!

**Review by nercesk — APPROVED** [review]

**Pull request merged**